plugins/argocd/src/utils.ts

```diff
@@ -59,3 +59,49 @@ export function getCustomK8sApi(): CustomObjectsApi {
   customK8sApi = customK8sApi ?? getClient().makeApiClient(CustomObjectsApi)
   return customK8sApi
 }
+
+/**
+ * Take a list of list of kubernetes resources and and ditch duplicates by name
```

Comment:
Plutôt `Take a list of kubernetes resources and remove duplicates by name.` ?

Aussi est-ce qu'il ne vaudrait pas mieux faire quelques chose du style :

```ts

export function uniqueResource<T extends { metadata: { name: string } }>(...lists: T[][]): T[] {
  return lists
    .flat()
    .reduce((acc, cur) => {
      if (acc.some(item => item.metadata.name === cur.metadata.name)) return acc
      return [...acc, cur]
    }, [] as T[])
}
```

Reply:
ça marche aussi très bien en effet !